xtrack/beam_elements/elements.py

```diff
@@ -1987,9 +1993,12 @@ def __init__(self, length=None, qx=0, qy=0,
             assert voltage_rf is None
             assert frequency_rf is None
             assert lag_rf is None
+            if bucket_length == None:
```

Comment:
(if we keep bucket length)
We need to check that, if it is provided, we are in the right longitudinal mode. Otherwise we need to raise an exception

Reply:
Done, the bucket_length is kept only for the mode 'fixed_qs', an error is raised otherise.